### PR TITLE
Add Users Page

### PR DESCRIPTION
**Pull request by briaguya**
image
image
image
image

### CODE REVIEW

File: src/components/Dashboard/ManageUsersPage/EditUserControl.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  label,
+  children,
+  className,
+}) {
+  return (
+    <div className={classNames('EditUser__control', className)}>
+      <div className="EditUser__control-meta">
+        <div className="EditUser__control-label">{`${label}${
+          isRequired ? '*' : ''
+        }`}</div>
```
Comment (luqmaan) [inline src/components/Dashboard/ManageUsersPage/EditUserControl.js]: No need to make this change, but a simpler way to write this:

```js
<div className="EditUser__control-label">
  {label} {isRequired && '*'}
</div>
```

File: src/components/Dashboard/ManageUsersPage/EditUserControl.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      </div>
+      <div className="EditUser__control-children">{children}</div>
+    </div>
+  );
+}
+
+EditUserControl.propTypes = {
```
Comment (luqmaan) [inline src/components/Dashboard/ManageUsersPage/EditUserControl.js]: 👍

File: src/components/Dashboard/ManageUsersPage/AddUserPage.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.AddUser {
+  margin-left: 10%;
+  margin-right: 10%;
```
Comment (luqmaan) [inline src/components/Dashboard/ManageUsersPage/AddUserPage.scss]: Probably not a good idea to use percentages for margins or layout in general.

File: src/components/Dashboard/ManageUsersPage/AddUserPage.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    redirect: false,
+    showModal: false,
+    userAdded: false,
+    emailSent: false,
+    errorMessage: null,
+  };
+
+  componentDidCatch(err) {
+    console.error(err);
+    this.setState({ errorMessage: err.message });
```
Comment (luqmaan) [inline src/components/Dashboard/ManageUsersPage/AddUserPage.js]: 🎉

File: src/components/Dashboard/ManageUsersPage/AddUserPage.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
+import React, { Component } from 'react';
+import 'components/Dashboard/ManageUsersPage/AddUserPage.css';
+import EditUser from 'components/Dashboard/ManageUsersPage/EditUser';
+import { Redirect } from 'react-router-dom';
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
+import generator from 'generate-password';
+import AddUserModal from 'components/Dashboard/ManageUsersPage/AddUserModal';
```
Comment (luqmaan) [inline src/components/Dashboard/ManageUsersPage/AddUserPage.js]: Don't need to change this now, but grouping external and internal imports can make things easier to read.

E.g. 

```
import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import generator from 'generate-password';

import EditUser from 'components/Dashboard/ManageUsersPage/EditUser';
import AddUserModal from 'components/Dashboard/ManageUsersPage/AddUserModal';
import 'components/Dashboard/ManageUsersPage/AddUserPage.css';
```

File: src/components/Dashboard/ManageUsersPage/AddUserPage.scss (hunk 3 above)
Comment (briaguya) [reply src/components/Dashboard/ManageUsersPage/AddUserPage.scss]: We should probably look through the codebase and find places where this is happening and clean it up, I made an issue #229 for it

**Pull request merged**